src/briefcase/platforms/linux/appimage.py

```diff
@@ -157,49 +140,77 @@ def support_package_url(self, support_revision):
             + self.support_package_filename(support_revision)
         )
 
-    def _pip_requires(self, requires: List[str]):
-        """Convert the requirements list to an AppImage compatible format.
-
-        If running in no-docker mode, this returns the requirements as specified
-        by the user.
+    def _install_app_requirements(
+        self,
+        app: AppConfig,
+        requires: List[str],
+        app_packages_path: Path,
+    ):
+        """Install requirements for the app with pip.
 
-        If running in Docker mode, any local file requirements are converted into
-        a reference in the /requirements mount folder.
+        This method pre-compiles any requirement defined using a local path
+        reference into an sdist tarball. This will be used when installing under
+        Docker, as local file references can't be accessed in the Docker
+        container.
 
-        :param requires: The user-specified list of app requirements
-        :returns: The final list of requirement arguments to pass to pip
+        :param app: The app configuration
+        :param requires: The list of requirements to install
+        :param app_packages_path: The full path of the app_packages folder into
+            which requirements should be installed.
         """
-        original = super()._pip_requires(requires)
-        if not self.use_docker:
-            return original
-
-        final = []
-        for requirement in original:
+        # If we're re-building requirements, purge any pre-existing local
+        # requirements.
+        local_requirements_path = self.local_requirements_path(app)
+        if local_requirements_path.exists():
+            self.tools.shutil.rmtree(local_requirements_path)
+        self.tools.os.mkdir(local_requirements_path)
+
+        # Iterate over every requirements, looking for local references
+        for requirement in requires:
             if _is_local_requirement(requirement):
-                source = Path(requirement).resolve()
-                final.append(f"/requirements/{source.name}")
-            else:
-                # Use the requirement as-specified
-                final.append(requirement)
-
-        return final
+                # Build an sdist for the local requirement
```

Comment:
`_is_local_requirement` will also return true for existing sdist or wheel filenames. Attempting to pass these to `build` doesn't make any sense, and will fail with the error `<filename> is not a directory`.

The easiest way to pass these files into Docker is probably to copy them into the same directory as the sdists we build ourselves.

Reply:
Good catch. Copying non-folders to the requirements folder makes sense; I'll do that.